runtime/doc/lua.txt

```diff
@@ -1581,4 +1581,71 @@ uri_to_fname({uri})                                       *vim.uri_to_fname()*
                 Return: ~
                     Filename
 
+==============================================================================
+Lua module: diff                                                    *lua-diff*
+
+diff({a}, {b}, {opts})                                            *vim.diff()*
+                Run diff on strings {a} and {b}. Any indices returned by this
+                function, either directly or via callback arguments, are
+                1-based.
+
+                Examples: >
+                vim.diff('a\n', 'b\nc\n')
+                -->
+                @@ -1 +1,2 @@
+                -a
+                +b
+                +c
+
+                vim.diff('a\n', 'b\nc\n', {hunk_lines = true})
+                -->
+                {
+                    {1, 1, 1, 2}
+                }
+<
+                Parameters: ~
+                    {a}      First string to compare
+                    {b}      Second string to compare
+                    {opts}   Optional parameters:
+                             • `on_hunk` (callback):
+                               Invoked for each hunk in the diff. Return a
+                               negative number to cancel the callback for any
+                               remaining hunks.
+                               Args:
+                               • `start_a` (integer): Start line of hunk in {a}.
+                               • `count_a` (integer): Hunk size in {a}.
+                               • `start_b` (integer): Start line of hunk in {b}.
+                               • `count_b` (integer): Hunk size in {b}.
+                             • `result_type` (string): Form of the returned diff:
+                               • "unified": (default) String in unified format.
+                               • "indices": Array of hunk locations.
+                               Note this option is ignored if `on_hunk` is
```

Comment:
should probably be an error rather than silently ignored?

Reply:
I changed it from an error because now we have `result_type` as an enum with an implicit default value of `unified` and I don't think we should error on this value whether it is implicit or explicit. And if we don't error for that then we shouldn't error for any other enum value.

When the option was a boolean/switch, the error was more justified as it is reasonable to expect the presence of other flags to be incompatible with each other.